### PR TITLE
adding firehose and eventbridge rule

### PR DESCRIPTION
**Pull request by xpositivityx**

### CODE REVIEW

File: aws/aws-kinesis-firehose/main.tf [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  destination = local.destinations[var.destination]
+
+  dynamic "extended_s3_configuration" {
+    for_each = var.dynamic_partitioning_enabled ? [1] : []
+    content {
+      role_arn            = aws_iam_role.firehose_role.arn
+      bucket_arn          = var.bucket_arn
+      prefix              = local.key_prefix
+      error_output_prefix = "errors/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/!{firehose:error-output-type}/"
+      buffer_size         = var.buffer_size     # In MB
```
Comment (WillBeebe) [inline aws/aws-kinesis-firehose/main.tf:24]: nit: for these params in the massdriver.yaml we've been adding the unit as a suffix. `buffer_size_mb`. What are your thoughts on doing that here?

File: aws/aws-kinesis-firehose/main.tf [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      buffer_interval     = var.buffer_interval # In seconds
```
Comment (WillBeebe) [inline aws/aws-kinesis-firehose/main.tf:25]: nit `buffer_interval_seconds` 🤷‍♂️

File: aws/aws-kinesis-firehose/main.tf [hunk 3]
```diff
@@ -0,0 +1,72 @@
+locals {
+  destinations = {
+    "s3" = "extended_s3"
+  }
+
+  default_key_prefix     = "data/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
+  partitioned_key_prefix = "data/!{partitionKeyFromQuery:partition_key}/!{timestamp:yyyy}/!{timestamp:MM}/!{timestamp:dd}/!{timestamp:HH}/"
```
Comment (WillBeebe) [inline aws/aws-kinesis-firehose/main.tf:7]: this is cool

File: aws/aws-kinesis-firehose/main.tf [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            parameter_value = format("{partition_key:%s}", var.query)
+          }
+        }
+      }
+    }
+  }
+}
+
+resource "aws_iam_role" "firehose_role" {
+  name        = "firehose_test_role"
```
Comment (WillBeebe) [inline aws/aws-kinesis-firehose/main.tf:51]:
```suggestion
  name        = "${local.name}-role"
```

File: aws/aws-eventbridge-rule/_variables.tf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  }
+}
+
+variable "event_filter_pattern" {
+  description = "Filter pattern to use in choosing which events to send"
+  type        = string
+  default     = "{}"
+
+  validation {
+    condition     = can(jsonencode(var.event_filter_pattern))
```
Comment (WillBeebe) [inline aws/aws-eventbridge-rule/_variables.tf:22]: 💯

**Review by WillBeebe — APPROVED** [review]
One small thing to change, otherwise LGTM

File: aws/aws-kinesis-firehose/main.tf (hunk 1 above)
Comment (coryodaniel) [reply aws/aws-kinesis-firehose/main.tf:24]: Fan of that over comments

File: aws/aws-kinesis-firehose/main.tf (hunk 4 above)
Comment (coryodaniel) [reply aws/aws-kinesis-firehose/main.tf:51]: I’d say leave off “role” you know it’s a role by the ARN and resource types. We would be using a suffix to distinguish between N roles in the same bundle/mod

File: aws/aws-kinesis-firehose/_variables.tf [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+variable "buffer_interval" {
+  description = "Amount of time in seconds in which to send data to the destination if the buffer size has not been fulfilled"
+  type        = number
+  default     = 300
+}
+
+variable "query" {
+  description = "JQ query which can extract a partition key from events"
```
Comment (coryodaniel) [inline aws/aws-kinesis-firehose/_variables.tf:30]: Does aws handle the query parsing or do we need to include our JQ provider?

Comment (xpositivityx) [reply aws/aws-kinesis-firehose/_variables.tf:30]: AWS handles it. you just have to pass it a query.

**Review by chrisghill — APPROVED** [review]

**Pull request merged**